### PR TITLE
chore: add plugin test workflow

### PR DESCRIPTION
**Pull request by Naturalclar**
Part of https://github.com/asdf-community/.github/issues/2

Added github actions workflow for plugin tests

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -73,3 +73,16 @@ jobs:
...
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: asdf_plugin_test
+        uses: asdf-vm/actions/plugin-test@v1.0.0
+        with:
+          command: direnv version
```
Comment (imbsky) [inline .github/workflows/workflow.yml]:
```suggestion
          command: alp --version
```

**Comment by imbsky** [comment]
It seems good overall! The version check command is incorrect, so please fix it.

**Comment by imbsky** [comment]
Oops, and could you add this to plugin_test job?

```yml
env:
  GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by imbsky** [comment]
@vic Hmm... Any idea?

```
FAILED: Incorrect permissions on /tmp/asdf.Xc8l/plugins/alp/bin/install. Must be executable by asdf
##[error]Action failed with error Error: The process 'asdf' failed with exit code 1
##[error]Node run failed with exit code 1
```

**Comment by vic** [comment]
> @vic Hmm... Any idea?
> 
> ```
> FAILED: Incorrect permissions on /tmp/asdf.Xc8l/plugins/alp/bin/install. Must be executable by asdf
> ##[error]Action failed with error Error: The process 'asdf' failed with exit code 1
> ##[error]Node run failed with exit code 1
> ```

Hm, is that file executable ? `chmod +x`

**Comment by imbsky** [comment]
Yes, alp binary is executable.

**Comment by imbsky** [comment]
This seems to be an error with the install script.

**Comment by imbsky** [comment]
Related issue: https://github.com/asdf-vm/asdf/issues/373

**Comment by imbsky** [comment]
Thanks @Naturalclar for your contribution!

**Comment by imbsky** [comment]
@all-contributors please add @Naturalclar for infra

**Pull request merged**